src/pfe/portal/modules/Templates.js

```diff
@@ -325,7 +326,8 @@ async function readRepoTemplatesJSON(repository) {
   try {
     const templateDetails = JSON.parse(res.body);
     for (const prop of ['name', 'description']) {
-      if (templateDetails.hasOwnProperty(prop)) {
+      // Only set the name or description of the repo if not given by the user
+      if (templateDetails.hasOwnProperty(prop) && !repository[prop]) {
```

Comment:
this should work, though performance could be improved if the `!repository[prop]` check were done before GETting the templates.json, to skip that request when name and desc are already set?

Reply:
What if only one is set?

Reply:
in that case we still need to get the one that isn't set by the user, right?

Reply:
Sorry, yeah understood what you meant so deleted the comment